# Workflow: 

- Step 1.  For 'Position_clean' column, find all rows that contain substring combo ('intern' OR 'co-op) AND ('data science' OR 'data scientist' OR 'ML' or 'machine learning') *case insensitive*
    - replace all with 'Data Scientist Intern' 
- Step 2.  find all rows that contain substring 'data science'*case insensitive*
    - replace case by case with 'xx Data Scientist' 
- Step 3. find all rows that contain substring 'analytics' *case insensitive*
    - replace case by case with 'xx Data Scientist' 
- Step 4. find all rows that contain substring 'ML' or 'machine learning' (assume: exclude 'AI' roles) *case-insensitive*
    - replace case by case with 'xx Data Scientist' 
 
- Step 5. Other special cases (from examining df.Position. unique()): replace case by case with 'xx Data Scientist' 
  
- Step 6: find all rows that do NOT contain substring *case-insensitive* 'Data Scientist'(e.g. 'software engineer', 'AI research scientist', 'data engineer', 'software developer', 'database developer', 'data analyst') -> replace with np.nan (for easier removal of these rows later) 


[see here for str.replace](https://stackoverflow.com/questions/39768547/replace-whole-string-if-it-contains-substring-in-pandas)

**note: order matters! e.g. '*Senior* *director*, data scientist' - need to change to LDS, before SR**

- Step 7A. replace any substrings 'lead' or 'director' or 'manager' or 'principal' -> 'Lead Data Scientist'
- Step 7B. replace any substrings 'sr' or 'senior' -> 'Senior Data Scientist'
- Step 7C. replace any substrings jr' or 'junior' -> 'Junior Data Scientist'
- Step 7D. replace substrings NOT 'intern' NOT 'junior' NOT 'senior' NOT 'lead' NOT NaN  -> 'Data Scientist' 


AFTER merging with Sasha's cleaned columns, do following: 
- drop rows that have NaN in the 'Position_clean' column [see how here](https://stackoverflow.com/questions/13413590/how-to-drop-rows-of-pandas-dataframe-whose-value-in-a-certain-column-is-nan)
- check and drop duplicates
- drop other rows with certain NaN fields

In [1]:
import pandas as pd
import numpy as np

In [2]:
INTERN = 'Data Scientist Intern'
JR = 'Junior Data Scientist'
SR = 'Senior Data Scientist'
DS = 'Data Scientist'
LDS = 'Lead Data Scientist'
rm = np.nan

In [3]:
df_uk = pd.read_csv('jack_uk.csv', index_col = 0)

In [4]:
df_uk.shape

(1048, 12)

In [5]:
# duplicate
df_uk['Position_clean'] = df_uk.loc[:, 'Position']

In [6]:
df_uk.head()

Company  \
0                  OpenMarket   
1                   causaLens   
2         Northrop Grumman UK   
3                 LGC Limited   
4  Neilson Financial Services   

                                         Description  \
0  We’ve got a pretty good thing going on here at...   
1  Data Scientist - Applied Science\r\r\r\r\r\r\r...   
2  Are you an experienced Data Scientist looking ...   
3  We are looking for an Immunogenicity Regulated...   
4  Data Scientist - Decision Science\r\r\r\r\r\r\...   

   Glassdoor Salary Estimate                                 Industry  \
0                        NaN  Enterprise Software & Network Solutions   
1                        NaN  Enterprise Software & Network Solutions   
2                        NaN                      Aerospace & Defense   
3                        NaN                                      NaN   
4                        NaN          Insurance Agencies & Brokerages   

                                            Location  \
0                                    London, England   
1                                    London, England   
2                                Cheltenham, England   
3  Fordham, Cambridgeshire, East of England, England   
4     Slough, Berkshire, South East England, England   

                                            Position  Rating  \
0                           Principal Data Scientist     4.5   
1                   Data Scientist - Applied Science     4.5   
2                           Principal Data Scientist     3.8   
3  Senior Scientist/Regulated Project Manager - I...     NaN   
4                                     Data Scientist     3.1   

                      Revenue                  Sector                  Size  \
0  $100 to $500 million (USD)  Information Technology  201 to 500 Employees   
1    Unknown / Non-Applicable  Information Technology     1 to 50 Employees   
2          $10+ billion (USD)     Aerospace & Defense      10000+ Employees   
3  Less than $1 million (USD)                     NaN               Unknown   
4    $10 to $25 million (USD)               Insurance   51 to 200 Employees   

                             Type  Year Founded  \
0  Subsidiary or Business Segment        1999.0   
1               Company - Private        2017.0   
2                Company - Public        1939.0   
3                Company - Public           NaN   
4               Company - Private        2012.0   

                                      Position_clean  
0                           Principal Data Scientist  
1                   Data Scientist - Applied Science  
2                           Principal Data Scientist  
3  Senior Scientist/Regulated Project Manager - I...  
4                                     Data Scientist

In [7]:
df_uk['Position'].unique()

array(['Principal Data Scientist', 'Data Scientist - Applied Science',
       'Senior Scientist/Regulated Project Manager - Immunogenicity',
       'Data Scientist',
       'Research Scientist Peri-and-Post Approval Studies (PPAS)',
       'Junior Data Scientist', 'Research Scientist Evidence Synthesis',
       'Senior Data Scientist',
       'Research Associate III/Senior Research Associate Data Collection Peri- and Post Approval Studies',
       'Junior Data Scientist - London',
       'Data Scientist - Engineering and Product',
       'Data Scientist - Reinforcement Learning',
       'Experienced Physical Sciences Scientist',
       'Research Data Scientist, Intern - Infrastructure Strategy',
       'Lead Scientist - Next Generation Sequencing',
       'Junior Data Scientist - Home based', 'Data Scientist Intern',
       'Data Science - Deep Learning Research',
       'Artificial Intelligence – Data Scientist',
       'Data Scientist (Menu Tribe)', 'Lead Data Scientist',
       'Dat

In [8]:
# Step 1. DS intern 
df_uk.loc[df_uk['Position'] == 'Internal only -Head Data Scientist, HateLab, School of Social Sciences', 'Position_clean'] = LDS
df_uk.loc[df_uk['Position'] == 'International Data Scientist', 'Position_clean'] = DS
# placement|intern|co-op
# BMW Plant Hams Hall, Birmingham - Quality Data Scientist - 13 Month Placement
# GE Aviation Data Science - 12 Month Placement
# GE Aviation Data Science - 12 Month Placement

In [9]:
s = ['intern', 'co-op', 'co op', 'placem']
cond_or1 = '|'.join(s)
t = ["data scien", "machine learn"]
cond_or2 = '|'.join(t)
bool_intern = (df_uk['Position_clean'].str.contains(cond_or1, case = False))& \
               ( (df_uk['Position_clean'].str.contains(cond_or2, case = False)) |\
                (df_uk['Position_clean'].str.contains('ML', case = True))
               ) 
df_uk[bool_intern] # display rows

Company  \
19                    Facebook   
24                        RELX   
32   LexisNexis Risk Solutions   
501               BMW Group UK   
799               BMW Group UK   
834           General Electric   
835                GE Aviation   

                                           Description  \
19   Facebook's mission is to give people the power...   
24   Undergraduate Data Scientists Industrial Place...   
32   Undergraduate Data Scientists Industrial Place...   
501  We believe in creating an environment where ou...   
799  We believe in creating an environment where ou...   
834  Job Description Summary\r\r\r\r\r\r\r\r\r\nIn ...   
835  Job Description Summary\r\r\r\r\r\r\r\r\r\nIn ...   

     Glassdoor Salary Estimate                                 Industry  \
19                         NaN                                 Internet   
24                         NaN  Enterprise Software & Network Solutions   
32                         NaN                               Consulting   
501                        NaN   Transportation Equipment Manufacturing   
799                        NaN   Transportation Equipment Manufacturing   
834                        NaN                 Industrial Manufacturing   
835                        NaN                      Aerospace & Defense   

                            Location  \
19                   London, England   
24                   London, England   
32                   London, England   
501  Warwick, West Midlands, England   
799  Warwick, West Midlands, England   
834               Eastleigh, England   
835               Eastleigh, England   

                                              Position  Rating  \
19   Research Data Scientist, Intern - Infrastructu...     4.3   
24                               Data Scientist Intern     3.7   
32                               Data Scientist Intern     3.8   
501  BMW Plant Hams Hall, Birmingham - Quality Data...     4.1   
799  BMW Plant Hams Hall, Birmingham - Quality Data...     4.1   
834  GE Aviation Data Science - 12 Month Placement,...     3.9   
835      GE Aviation Data Science - 12 Month Placement     3.7   

                              Revenue                  Sector  \
19            $5 to $10 billion (USD)  Information Technology   
24                 $10+ billion (USD)  Information Technology   
32             $2 to $5 billion (USD)       Business Services   
501          Unknown / Non-Applicable           Manufacturing   
799          Unknown / Non-Applicable           Manufacturing   
834                $10+ billion (USD)           Manufacturing   
835  $500 million to $1 billion (USD)     Aerospace & Defense   

                        Size                            Type  Year Founded  \
19          10000+ Employees                Company - Public        2004.0   
24          10000+ Employees                Company - Public        1880.0   
32    1001 to 5000 Employees                Company - Public        2000.0   
501  5001 to 10000 Employees  Subsidiary or Business Segment           NaN   
799  5001 to 10000 Employees  Subsidiary or Business Segment           NaN   
834         10000+ Employees                Company - Public        1892.0   
835   1001 to 5000 Employees  Subsidiary or Business Segment        1917.0   

                                        Position_clean  
19   Research Data Scientist, Intern - Infrastructu...  
24                               Data Scientist Intern  
32                               Data Scientist Intern  
501  BMW Plant Hams Hall, Birmingham - Quality Data...  
799  BMW Plant Hams Hall, Birmingham - Quality Data...  
834  GE Aviation Data Science - 12 Month Placement,...  
835      GE Aviation Data Science - 12 Month Placement

In [10]:
df_uk.loc[bool_intern, 'Position_clean'] = INTERN
# check
df_uk[bool_intern][['Position', 'Position_clean']]

Position         Position_clean
19   Research Data Scientist, Intern - Infrastructu...  Data Scientist Intern
24                               Data Scientist Intern  Data Scientist Intern
32                               Data Scientist Intern  Data Scientist Intern
501  BMW Plant Hams Hall, Birmingham - Quality Data...  Data Scientist Intern
799  BMW Plant Hams Hall, Birmingham - Quality Data...  Data Scientist Intern
834  GE Aviation Data Science - 12 Month Placement,...  Data Scientist Intern
835      GE Aviation Data Science - 12 Month Placement  Data Scientist Intern

In [11]:
## Step 2
bool_2 = df_uk['Position'].str.contains('data science', case = False)
df_uk[bool_2]['Position'].unique() # display

array(['Data Science - Deep Learning Research',
       'Data Science - Machine Learning Research',
       'Data Science Engineer', 'Data Science Manager',
       'Head of Life Data Science', 'Data Science Lead, Financial Crime',
       'Data Science Lead, Reliability Analytics', 'Data Science',
       'Senior Strategic Consultant - Data Science & Analytics',
       'Senior Data Science Manager',
       'Senior Data Science Manager - Strategic Analytics',
       'Data Science Campus - Programme Manager',
       'GE Aviation Data Science - 12 Month Placement, Eastleigh',
       'GE Aviation Data Science - 12 Month Placement', 'VP Data Science',
       'VP of Data Science', 'Senior Software Engineer - Data Science',
       'Head of Data Science', 'Head of Data Science Engineering',
       'Lead Data Scientist / Data Science Director',
       'Sr. Manager, Data Science',
       'Director of Data Products and Data Science',
       'Director of Decision & Data Science', 'Data Science Directo

In [12]:
# remove SE eng
df_uk.loc[df_uk['Position'] == 'Senior Software Engineer - Data Science', 'Position_clean'] = rm

In [13]:
# Data Science
df_uk.loc[df_uk['Position'] == 'Data Science', 'Position_clean'] = DS

In [14]:
# lead ds
bool_lds = (df_uk['Position_clean'].str.contains('VP|head|manager|direct|lead|principal', case = False)) & \
           (df_uk['Position_clean'].str.contains('data science', case = False))
df_uk.loc[bool_lds, 'Position_clean'] = LDS
# sr ds
bool_sr = (df_uk['Position_clean'].str.contains('senior|sr', case = False)) & \
           (df_uk['Position_clean'].str.contains('data science', case = False))
df_uk.loc[bool_sr, 'Position_clean'] = SR
# ds
bool_ds = (df_uk['Position_clean'].str.contains('engineer|learning res', case = False)) & \
           (df_uk['Position_clean'].str.contains('data science', case = False))
df_uk.loc[bool_ds, 'Position_clean'] = DS

In [15]:
# chceck
df_uk[bool_2][['Position','Position_clean']]

Position         Position_clean
25               Data Science - Deep Learning Research         Data Scientist
33            Data Science - Machine Learning Research         Data Scientist
86                               Data Science Engineer         Data Scientist
115                               Data Science Manager    Lead Data Scientist
127                               Data Science Manager    Lead Data Scientist
135                          Head of Life Data Science    Lead Data Scientist
152                               Data Science Manager    Lead Data Scientist
331                               Data Science Manager    Lead Data Scientist
352                 Data Science Lead, Financial Crime    Lead Data Scientist
382           Data Science Lead, Reliability Analytics    Lead Data Scientist
406                                       Data Science         Data Scientist
436  Senior Strategic Consultant - Data Science & A...  Senior Data Scientist
447                        Senior Data Science Manager    Lead Data Scientist
460  Senior Data Science Manager - Strategic Analytics    Lead Data Scientist
464            Data Science Campus - Programme Manager    Lead Data Scientist
497                               Data Science Manager    Lead Data Scientist
498                               Data Science Manager    Lead Data Scientist
500           Data Science - Machine Learning Research         Data Scientist
507              Data Science - Deep Learning Research         Data Scientist
508                               Data Science Manager    Lead Data Scientist
630                               Data Science Manager    Lead Data Scientist
653                 Data Science Lead, Financial Crime    Lead Data Scientist
683           Data Science Lead, Reliability Analytics    Lead Data Scientist
705                                       Data Science         Data Scientist
736  Senior Strategic Consultant - Data Science & A...  Senior Data Scientist
747                        Senior Data Science Manager    Lead Data Scientist
759  Senior Data Science Manager - Strategic Analytics    Lead Data Scientist
763            Data Science Campus - Programme Manager    Lead Data Scientist
797                               Data Science Manager    Lead Data Scientist
798                               Data Science Manager    Lead Data Scientist
800           Data Science - Machine Learning Research         Data Scientist
806              Data Science - Deep Learning Research         Data Scientist
808                               Data Science Manager    Lead Data Scientist
821                               Data Science Manager    Lead Data Scientist
830                          Head of Life Data Science    Lead Data Scientist
834  GE Aviation Data Science - 12 Month Placement,...  Data Scientist Intern
835      GE Aviation Data Science - 12 Month Placement  Data Scientist Intern
836                                    VP Data Science    Lead Data Scientist
837                                 VP of Data Science    Lead Data Scientist
839            Senior Software Engineer - Data Science                    NaN
841                               Head of Data Science    Lead Data Scientist
847                               Head of Data Science    Lead Data Scientist
849                   Head of Data Science Engineering    Lead Data Scientist
851        Lead Data Scientist / Data Science Director    Lead Data Scientist
859                   Head of Data Science Engineering    Lead Data Scientist
864                               Data Science Manager    Lead Data Scientist
867                          Sr. Manager, Data Science    Lead Data Scientist
887                               Head of Data Science    Lead Data Scientist
907         Director of Data Products and Data Science    Lead Data Scientist
951                Director of Decision & Data Science    Lead Data Scientist
958                              Data Science Director    Lead Data 

In [16]:
## step 3
bool_3 = df_uk['Position'].str.contains('analytics', case = False)
df_uk[bool_3]['Position'].unique()

array(['Data & Analytics Consultant',
       'Data & Analytics Consultant (m/f/d)',
       'Research Scientist – Analytics Group',
       'Senior Product Manager (Data Visualisation / Analytics)',
       'Senior Data Scientist- (Machine Learning & Advanced Analytics)',
       'Senior Analyst - Strategy & Analytics',
       'Customer Analytics Data Scientist - Senior Associate',
       'Senior Data Scientist, Product Analytics (Belfast, UK)',
       'Head of Data Analytics',
       'Senior Data Scientist, Advanced Analytics',
       'Data Science Lead, Reliability Analytics',
       'Senior Strategic Consultant - Data Science & Analytics',
       'Senior Data Science Manager - Strategic Analytics',
       'Data Scientist – Customer Analytics',
       'Senior Data Analytics Consultant'], dtype=object)

In [17]:
# remove PM
df_uk.loc[df_uk['Position'] == 'Senior Product Manager (Data Visualisation / Analytics)', 'Position_clean'] = rm

In [18]:
# lead ds
bool_lds = (df_uk['Position_clean'].str.contains('VP|head|manag|director|lead|principal', case = False, na = False)) & \
           (df_uk['Position_clean'].str.contains('analytics', case = False, na = False))
df_uk.loc[bool_lds, 'Position_clean'] = LDS
# sr ds
bool_sr = (df_uk['Position_clean'].str.contains('senior|sr', case = False, na = False)) & \
           (df_uk['Position_clean'].str.contains('analytics', case = False, na = False))
df_uk.loc[bool_sr, 'Position_clean'] = SR
# ds
bool_ds = df_uk['Position_clean'].str.contains('analytics', case = False, na = False)
df_uk.loc[bool_ds, 'Position_clean'] = DS

In [19]:
# check 
df_uk[bool_3][['Position', 'Position_clean']]

Position         Position_clean
48                         Data & Analytics Consultant         Data Scientist
98                 Data & Analytics Consultant (m/f/d)         Data Scientist
125               Research Scientist – Analytics Group         Data Scientist
154  Senior Product Manager (Data Visualisation / A...                    NaN
227  Senior Data Scientist- (Machine Learning & Adv...  Senior Data Scientist
234              Senior Analyst - Strategy & Analytics  Senior Data Scientist
335  Customer Analytics Data Scientist - Senior Ass...  Senior Data Scientist
337  Senior Data Scientist, Product Analytics (Belf...  Senior Data Scientist
344                             Head of Data Analytics    Lead Data Scientist
363  Senior Data Scientist, Product Analytics (Belf...  Senior Data Scientist
367          Senior Data Scientist, Advanced Analytics  Senior Data Scientist
382           Data Science Lead, Reliability Analytics    Lead Data Scientist
436  Senior Strategic Consultant - Data Science & A...  Senior Data Scientist
460  Senior Data Science Manager - Strategic Analytics    Lead Data Scientist
465                Data Scientist – Customer Analytics         Data Scientist
527  Senior Data Scientist- (Machine Learning & Adv...  Senior Data Scientist
533              Senior Analyst - Strategy & Analytics  Senior Data Scientist
635  Customer Analytics Data Scientist - Senior Ass...  Senior Data Scientist
636  Senior Data Scientist, Product Analytics (Belf...  Senior Data Scientist
646                             Head of Data Analytics    Lead Data Scientist
662  Senior Data Scientist, Product Analytics (Belf...  Senior Data Scientist
666          Senior Data Scientist, Advanced Analytics  Senior Data Scientist
683           Data Science Lead, Reliability Analytics    Lead Data Scientist
736  Senior Strategic Consultant - Data Science & A...  Senior Data Scientist
759  Senior Data Science Manager - Strategic Analytics    Lead Data Scientist
766                Data Scientist – Customer Analytics         Data Scientist
862                   Senior Data Analytics Consultant  Senior Data Scientist
975  Senior Data Scientist- (Machine Learning & Adv...  Senior Data Scientist
986              Senior Analyst - Strategy & Analytics  Senior Data Scientist

In [20]:
## Step 4 - no need 
bool_ml =  df_uk['Position'].str.contains('data', case = False, na = False)&\
            (df_uk['Position'].str.contains('machine learn', case = False, na = False)|\
         df_uk['Position'].str.contains('ML', case = True, na = False))
df_uk[bool_ml][['Position', 'Position_clean']]

Position  \
33             Data Science - Machine Learning Research   
64                    Senior Data Scientist/ML Engineer   
130            Senior Data Scientist - Machine Learning   
182                        Sr. Data Scientist - ML Labs   
199                        Sr. Data Scientist - ML Labs   
227   Senior Data Scientist- (Machine Learning & Adv...   
244                   Senior Data Scientist/ML Engineer   
370   Senior Data Scientist/ Machine Learning Developer   
372                 Senior Data Scientist / ML Engineer   
377            Senior Data Scientist - Machine Learning   
409                        Sr. Data Scientist - ML Labs   
500            Data Science - Machine Learning Research   
527   Senior Data Scientist- (Machine Learning & Adv...   
546                   Senior Data Scientist/ML Engineer   
669   Senior Data Scientist/ Machine Learning Developer   
673                 Senior Data Scientist / ML Engineer   
677            Senior Data Scientist - Machine Learning   
707                        Sr. Data Scientist - ML Labs   
800            Data Science - Machine Learning Research   
975   Senior Data Scientist- (Machine Learning & Adv...   
1018                  Senior Data Scientist/ML Engineer   

                                         Position_clean  
33                                       Data Scientist  
64                    Senior Data Scientist/ML Engineer  
130            Senior Data Scientist - Machine Learning  
182                        Sr. Data Scientist - ML Labs  
199                        Sr. Data Scientist - ML Labs  
227                               Senior Data Scientist  
244                   Senior Data Scientist/ML Engineer  
370   Senior Data Scientist/ Machine Learning Developer  
372                 Senior Data Scientist / ML Engineer  
377            Senior Data Scientist - Machine Learning  
409                        Sr. Data Scientist - ML Labs  
500                                      Data Scientist  
527                               Senior Data Scientist  
546                   Senior Data Scientist/ML Engineer  
669   Senior Data Scientist/ Machine Learning Developer  
673                 Senior Data Scientist / ML Engineer  
677            Senior Data Scientist - Machine Learning  
707                        Sr. Data Scientist - ML Labs  
800                                      Data Scientist  
975                               Senior Data Scientist  
1018                  Senior Data Scientist/ML Engineer

In [21]:
## Step 5 
df_uk.loc[df_uk['Position'] == 'Senior Data & Applied Scientist', 'Position_clean'] = SR
df_uk.loc[df_uk['Position'] == 'Senior/Principal Statistician, Statistical Innovation', 'Position_clean'] = SR

In [22]:
bool_dec1 = df_uk['Position_clean'].str.contains('decision scien', case = False, na = False ) &\
            df_uk['Position_clean'].str.contains('principal|lead|head|manage|direct|VP', case = False, na = False )
bool_dec2 = df_uk['Position_clean'].str.contains('decision scien', case = False, na = False ) &\
            df_uk['Position_clean'].str.contains('senior|sr', case = False, na = False )

df_uk.loc[bool_dec1, 'Position_clean'] = LDS
df_uk.loc[bool_dec2, 'Position_clean'] = SR

In [23]:
bool_spec1 = df_uk['Position_clean'].str.contains('head of data|data insight consultant', case = False, na = False )
bool_spec2 = df_uk['Position_clean'].str.contains('data driven|Data Analyst/Sci', case = False, na = False )

df_uk.loc[bool_spec1, 'Position_clean'] = LDS
df_uk.loc[bool_spec2, 'Position_clean'] = SR

In [24]:
## Step 6 
bool_notDS = df_uk['Position_clean'].str.contains('Data Scientist', case = False) == False
#df_uk[bool_notDS]['Position_clean'].unique()
df_uk.loc[bool_notDS, 'Position_clean'] = rm

In [25]:
# check no. of unique data scientist listings left after removing irrelevant rows
df_uk[df_uk['Position_clean'].str.contains('Data Scientist', case = False) == True]['Position'].unique().shape

(133,)

In [26]:
df_uk['Position'].unique().shape

(336,)

## Out of 336 *unique* UK listings, get 133 *unique* data scientist (includes: 5 intern,  3 Jr.,   42 Sr.,  40 Lead,   43 data scientist) listings

In [27]:
## Step 7 
bool_lds = df_uk['Position_clean'].str.contains('lead|principal|manage|direct|VP|head', case = False, na = False)
#df_ca[bool_lds]['Position'].unique().shape 
#df_uk[bool_lds]['Position'].unique()
df_uk.loc[bool_lds, 'Position_clean'] = LDS

In [28]:
bool_sr = df_uk['Position_clean'].str.contains('senior|sr', case = False, na = False)
#df_uk[bool_sr]['Position'].unique().shape
df_uk.loc[bool_sr, 'Position_clean'] = SR

In [33]:
bool_jr = df_uk['Position_clean'].str.contains('junior|jr|new grad|entry', case = False, na = False)
df_uk[bool_jr]['Position'].unique().shape 
df_uk.loc[bool_jr, 'Position_clean'] = JR

In [35]:
bool_remain = (df_uk['Position_clean'].str.contains('senior', case = False, na = False) == False) &\
                (df_uk['Position_clean'].str.contains('junior', case = False, na = False) == False) &\
                (df_uk['Position_clean'].str.contains('lead', case = False, na = False) == False) &\
                (df_uk['Position_clean'].str.contains('intern', case = False, na = False) == False) &\
                df_uk['Position_clean'].notna()
df_uk.loc[bool_remain, 'Position_clean'] = DS

In [37]:
# check 
df_uk['Position_clean'].unique()

array(['Lead Data Scientist', 'Data Scientist', nan,
       'Junior Data Scientist', 'Senior Data Scientist',
       'Data Scientist Intern'], dtype=object)

In [38]:
df_uk[['Position', 'Position_clean']]

Position         Position_clean
0                              Principal Data Scientist    Lead Data Scientist
1                      Data Scientist - Applied Science         Data Scientist
2                              Principal Data Scientist    Lead Data Scientist
3     Senior Scientist/Regulated Project Manager - I...                    NaN
4                                        Data Scientist         Data Scientist
5     Research Scientist Peri-and-Post Approval Stud...                    NaN
6                                 Junior Data Scientist  Junior Data Scientist
7                                 Junior Data Scientist  Junior Data Scientist
8                                 Junior Data Scientist  Junior Data Scientist
9                                 Junior Data Scientist  Junior Data Scientist
10                Research Scientist Evidence Synthesis                    NaN
11                                Senior Data Scientist  Senior Data Scientist
12    Research Associate III/Senior Research Associa...                    NaN
13                                Junior Data Scientist  Junior Data Scientist
14                       Junior Data Scientist - London  Junior Data Scientist
15             Data Scientist - Engineering and Product         Data Scientist
16                                       Data Scientist         Data Scientist
17              Data Scientist - Reinforcement Learning         Data Scientist
18              Experienced Physical Sciences Scientist                    NaN
19    Research Data Scientist, Intern - Infrastructu...  Data Scientist Intern
20          Lead Scientist - Next Generation Sequencing                    NaN
21                                       Data Scientist         Data Scientist
22                                       Data Scientist         Data Scientist
23                   Junior Data Scientist - Home based  Junior Data Scientist
24                                Data Scientist Intern  Data Scientist Intern
25                Data Science - Deep Learning Research         Data Scientist
26             Artificial Intelligence – Data Scientist         Data Scientist
27                          Data Scientist (Menu Tribe)         Data Scientist
28                                  Lead Data Scientist    Lead Data Scientist
29              Data Scientist / Operational Researcher         Data Scientist
...                                                 ...                    ...
1018                  Senior Data Scientist/ML Engineer  Senior Data Scientist
1019                                Lead Data Scientist    Lead Data Scientist
1020                              Senior Data Scientist  Senior Data Scientist
1021                              Senior Data Scientist  Senior Data Scientist
1022                              Senior Data Scientist  Senior Data Scientist
1023                              Senior Data Scientist  Senior Data Scientist
1024                              Senior Data Scientist  Senior Data Scientist
1025                              Senior Data Scientist  Senior Data Scientist
1026                              Senior Data Scientist  Senior Data Scientist
1027                                Lead Data Scientist    Lead Data Scientist
1028                              Senior Data Scientist  Senior Data Scientist
1029                           Principal Data Scientist    Lead Data Scientist
1030                                Lead Data Scientist    Lead Data Scientist
1031                              Senior Data Scientist  Senior Data Scientist
1032                              Senior Data Scientist  Senior Data Scientist
1033  Senior Frontend Developer - Motor Insurance ch...                    NaN
1034                              Senior Data Scientist  Senior Data Scientist
1035                              Senior Data Scientist  Senior Data Scientist
1036                              Senior Data Scientist  Senior Data Scientist
1037              

In [39]:
# write to csv
df_uk.to_csv('elisa_position_uk.csv')

In [40]:
# read in to check 
df = pd.read_csv('elisa_position_uk.csv', index_col = 0)
df.shape

(1048, 13)

In [45]:
df_uk[df_uk['Position'] == 'Data Scientist - Managed Rules']

Company                                        Description  \
450  Cloudflare  About Us\r\r\r\r\r\r\r\r\r\r\r\r\r\r\r\r\r\r\r...   
751  Cloudflare  About Us\r\r\r\r\r\r\r\r\r\r\r\n\r\r\r\r\r\r\r...   

     Glassdoor Salary Estimate  Industry         Location  \
450                        NaN  Internet  London, England   
751                        NaN  Internet  London, England   

                           Position  Rating                   Revenue  \
450  Data Scientist - Managed Rules     4.0  Unknown / Non-Applicable   
751  Data Scientist - Managed Rules     4.0  Unknown / Non-Applicable   

                     Sector                    Size              Type  \
450  Information Technology  1001 to 5000 Employees  Company - Public   
751  Information Technology  1001 to 5000 Employees  Company - Public   

     Year Founded       Position_clean  
450        2010.0  Lead Data Scientist  
751        2010.0  Lead Data Scientist